### PR TITLE
feat: 🎸 rendering witness result data object instead of formatted HTML

### PR DESCRIPTION
**Pull request by lobogang**
Screen Shot 2022-04-11 at 23 02 21
Screen Shot 2022-04-11 at 23 02 13

### CODE REVIEW

File: src/components/pages/OfflineVerification/PageVerificationInfo.tsx [hunk 1]
```diff
@@ -163,6 +163,7 @@ const PageVerificationInfo = ({
         doVerifyMerkleProof,
         null
       );
+      console.log({ verificationStatus, details });
```
Comment (rht) [inline src/components/pages/OfflineVerification/PageVerificationInfo.tsx:166]: Accidentally committed.

File: src/utils/formatPageInfo/index.ts [hunk 2]
```diff
@@ -5,13 +5,55 @@ import { formatDBTimestampToObject } from "./timestampFormatter";
 import type { SignatureData, Witness, SignatureStatus } from "../../types";
 import * as nameResolver from "../../name_resolver";
 
+type MerkleProofStatus = "VALID" | "INVALID" | "DOMAIN_SNAPSHOT";
+type EtherscanResult = "true" | "false" | "Transaction hash not found" | string;
+
+// let errMsg
```
Comment (rht) [inline src/utils/formatPageInfo/index.ts:11]: I think we should instead put a permalink here: https://github.com/inblockio/aqua-verifier-js/blob/de3432e7309c642159e756ddf7dc49e602ab660d/index.js#L346-L354.

File: src/utils/formatPageInfo/index.ts [hunk 3]
```diff
@@ -5,13 +5,55 @@ import { formatDBTimestampToObject } from "./timestampFormatter";
...
+  domain_snapshot_genesis_hash: string;
+  merkle_root: string;
+  witness_event_verification_hash: string;
+} | null;
+
+export type WitnessResultProps = {
+  actual_witness_event_verification_hash: string;
+  doVerifyMerkleProof: boolean;
+  etherscan_error_message: EtherscanErrorMessage;
+  etherscan_result: EtherscanResult;
```
Comment (rht) [inline src/utils/formatPageInfo/index.ts:42]: I think this is unused? I can't find a code that tells users to verify manually if this is "false".

File: src/components/VerificationLog/WitnessResult.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        title="Verification Hash"
+        content={actual_witness_event_verification_hash}
+        truncateContent
+      />
+      <Extra />
+      {etherscan_error_message && (
+        <ListItem>{etherscan_error_message}</ListItem>
+      )}
+      <ConditionalIconListItem
+        condition={merkle_proof_status === "VALID"}
```
Comment (rht) [inline src/components/VerificationLog/WitnessResult.tsx:122]: This is not taking into account when `merkle_proof_status` is "DOMAIN_SNAPSHOT".

**Comment by rht** [comment]
Still missing https://github.com/inblockio/aqua-verifier-js/blob/de3432e7309c642159e756ddf7dc49e602ab660d/index.js#L544-L564= ?

**Comment by FantasticoFox** [comment]
Looks like we need some form of visual grouping to make clear this is all witness data. Right now I could not tell.

File: src/utils/formatPageInfo/index.ts (hunk 2 above)
Comment (lobogang) [reply src/utils/formatPageInfo/index.ts:11]: I think we should add some typing or other documentation that is more reliable than a permalink. If the lines between 346-354 are ever occupied by other code then the linking will be meaningless. Can we consider adding some typing to the aqua verifier that is then exported and imported here?

File: src/components/pages/OfflineVerification/PageVerificationInfo.tsx (hunk 1 above)
Comment (lobogang) [reply src/components/pages/OfflineVerification/PageVerificationInfo.tsx:166]: thanks, will resolve in upcoming commit. This should be caught by a linter in the future.

File: src/utils/formatPageInfo/index.ts (hunk 3 above)
Comment (lobogang) [reply src/utils/formatPageInfo/index.ts:42]: fair enough, will remove

**Comment by lobogang** [comment]
> Still missing https://github.com/inblockio/aqua-verifier-js/blob/de3432e7309c642159e756ddf7dc49e602ab660d/index.js#L544-L564= ?

good point. I'll add that in an upcoming commit

File: src/components/VerificationLog/WitnessResult.tsx (hunk 4 above)
Comment (lobogang) [reply src/components/VerificationLog/WitnessResult.tsx]: added DOMAIN_SNAPSHOT to the condition, will handle rendering in another commit

**Comment by lobogang** [comment]
> Looks like we need some form of visual grouping to make clear this is all witness data. Right now I could not tell.

Do you have some ideas for how to do this? I think this is a place where some design would be useful. I only used the former HTML output and this documentation: https://github.com/inblockio/aqua-verifier-webextension/issues/47#issuecomment-1016208813. Perhaps finetuning the layout can be done in another PR once we're sure we have all the data rendering to the page correctly

File: src/components/pages/OfflineVerification/PageVerificationInfo.tsx (hunk 1 above)
Comment (lobogang) [reply src/components/pages/OfflineVerification/PageVerificationInfo.tsx]: made a note to add some linting. will handle it in this PR or in a small followup

File: src/utils/formatPageInfo/index.ts (hunk 2 above)
Comment (rht) [reply src/utils/formatPageInfo/index.ts:10]: That link is a permalink, which basically links to a specific state of the code of a specific commit. Compare that with https://github.com/inblockio/aqua-verifier-js/blob/main/index.js#L346-L353=, which is a dynamic link to the latest main.

Comment (rht) [reply src/utils/formatPageInfo/index.ts:10]: I haven't done anything to improve the aqua-verifier-js repo for a while due to the plan to use aqua-verifier-go via wasm instead.

Comment (lobogang) [reply src/utils/formatPageInfo/index.ts]: thx for clarification :) I've updated docs with permalinks

**Comment by rht** [comment]
I rebased this PR.

**Pull request merged**